Fuente: [https://www.kaggle.com/datasets/saurabhbadole/latest-data-science-job-salaries-2024?select=DataScience_salaries_2024.csv]

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [3]:
df = pd.read_csv(r'..\data\DataScience_salaries_2024.csv')

## INTRODUCCION

En el ámbito en constante evolución de la ciencia de datos, resulta esencial comprender las tendencias y patrones salariales para los profesionales y las organizaciones. El objetivo de este conjunto de datos es proporcionar una visión del panorama salarial en ciencia de datos desde 2020 hasta 2024. Al examinar estos datos a lo largo de este período, se busca permitir a quien desee obtener información valiosa sobre las tendencias salariales, las diferencias regionales y los factores potenciales que afectan la compensación en la comunidad de ciencia de datos. (lenguaje casual)
- basarse en los graficos y ahora necesito redactar una introduccion del trabajo para dar entendr la intencion de lo q querias hacer. me interes es saber cuanto gana un empleado segun diversos parametros variables (mi intencion es conocer distintas realidades de los trabajadores en el ambito de la ciencia de datos)

## METADATA

    work_year(int): El año de los datos relacionados con el salario laboral.The year of the data related to the job salary.
    experience_level (str): El nivel de experiencia del empleado (entry-level, mid-level, senior-level).
    employment_type (str): El tipo de empleo (full-time, part-time, por contrato).
    job_title (str): El título o rol del empleado dentro del campo de la ciencia de datos.
    salary (int) : Salario del empleado en moneda local.
    salary_currency (str): La moneda en la que se paga el salario.
    salary_in_usd (int): El salario convertido a dólares estadounidenses para estandarización.
    employee_residence (str): La ubicación de residencia del empleado.
    remote_ratio (int): La proporción de trabajo remoto permitida para el puesto.
    company_location (str): La ubicación de la empresa.
    company_size (str): El tamaño de la empresa basado en el número de empleados o los ingresos.


In [6]:
df.head()

work_year experience_level employment_type                       job_title  \
0       2021               MI              FT                  Data Scientist   
1       2021               MI              FT                 BI Data Analyst   
2       2020               MI              FT                  Data Scientist   
3       2021               MI              FT                     ML Engineer   
4       2022               SE              FT  Lead Machine Learning Engineer   

     salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0  30400000             CLP          40038                 CL           100   
1  11000000             HUF          36259                 HU            50   
2  11000000             HUF          35735                 HU            50   
3   8500000             JPY          77364                 JP            50   
4   7500000             INR          95386                 IN            50   

  company_location company_size  
0               CL            L  
1               US            L  
2               HU            L  
3               JP            S  
4               IN            L

In [8]:
df.shape #Dimensiones de la tabla (N°Filas,N°Columnas).

(14838, 11)

In [9]:
df.info() #Muestra columnas del DataFrame

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14838 entries, 0 to 14837
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           14838 non-null  int64 
 1   experience_level    14838 non-null  object
 2   employment_type     14838 non-null  object
 3   job_title           14838 non-null  object
 4   salary              14838 non-null  int64 
 5   salary_currency     14838 non-null  object
 6   salary_in_usd       14838 non-null  int64 
 7   employee_residence  14838 non-null  object
 8   remote_ratio        14838 non-null  int64 
 9   company_location    14838 non-null  object
 10  company_size        14838 non-null  object
dtypes: int64(4), object(7)
memory usage: 1.2+ MB


Como el archivo tiene 14838 filas
y no hay columnas que presenten menos
de 14838 celdas no nulas se concluye 
que no hay falta de informacion.

Como el tipo de dato coincide con lo esperado no se encontraron errores en este aspecto.

In [10]:
df.duplicated().sum() #Cuenta cuantas filas duplicadas hay en el Data Frame.

np.int64(5711)

Se han encontrado 5711 registros duplicados pero debido a la estructura misma del dataset se puede asumir que se tratan de personas distintas contratadas por la misma entrada, por el mismo cargo y por el mismo sueldo dado que no existe una llave primaria; se ha decidido conservar los registros duplicados. 

In [11]:
df['experience_level'].unique()

array(['MI', 'SE', 'EN', 'EX'], dtype=object)

#### Descripción de los niveles de experiencia
     EN = entry-level
     MI = mid-level 
     SE = senior-level
     EX= expert-level

In [12]:
df['employment_type'].unique()

array(['FT', 'FL', 'CT', 'PT'], dtype=object)

#### Descripción de tipos de contratos de empleo
FT = Full-time
FL = Free Lance
CT = contract
PT = Part-Time

In [13]:
df['job_title'].unique() 
# En esta columna no se encuentran errores de tipeo ni formato

array(['Data Scientist', 'BI Data Analyst', 'ML Engineer',
       'Lead Machine Learning Engineer', 'Data Science Manager',
       'Head of Machine Learning', 'Research Engineer',
       'Head of Data Science', 'AI Programmer',
       'Machine Learning Engineer', 'Lead Data Scientist',
       'Data Engineer', 'Applied Machine Learning Scientist',
       'Lead Data Analyst', 'Data Analytics Manager',
       'Data Integration Specialist', 'Principal Data Architect',
       'NLP Engineer', 'Big Data Engineer', 'AI Research Engineer',
       'Machine Learning Software Engineer', 'Data Analyst',
       'Applied Data Scientist', 'AI Scientist', 'Data Analytics Lead',
       'Business Data Analyst', 'Product Data Analyst',
       'Computer Vision Engineer', 'Data Science Consultant',
       'AI Architect', 'Analytics Engineer', 'Machine Learning Scientist',
       'Research Scientist', 'Prompt Engineer',
       'Principal Data Scientist', 'Applied Scientist',
       'Deep Learning Engineer', 

In [14]:
df['salary_currency'].unique() 

array(['CLP', 'HUF', 'JPY', 'INR', 'ILS', 'NOK', 'THB', 'PHP', 'USD',
       'MXN', 'ZAR', 'HKD', 'CAD', 'TRY', 'GBP', 'EUR', 'DKK', 'CHF',
       'NZD', 'PLN', 'BRL', 'AUD', 'SGD'], dtype=object)

#### Descripción de Divisas.
    AUD - Dólar australiano (Australian Dollar)
    BRL - Real brasileño (Brazilian Real)
    CAD - Dólar canadiense (Canadian Dollar)
    CHF - Franco suizo (Swiss Franc)
    CLP - Peso chileno (Chilean Peso)
    DKK - Corona danesa (Danish Krone)
    EUR - Euro (Euro)
    GBP - Libra esterlina (British Pound)
    HKD - Dólar de Hong Kong (Hong Kong Dollar)
    HUF - Florín húngaro (Hungarian Forint)
    ILS - Nuevo shéquel israelí (Israeli New Shekel)
    INR - Rupia india (Indian Rupee)
    JPY - Yen japonés (Japanese Yen)
    MXN - Peso mexicano (Mexican Peso)
    NOK - Corona noruega (Norwegian Krone)
    NZD - Dólar neozelandés (New Zealand Dollar)
    PHP - Peso filipino (Philippine Peso)
    PLN - Zloty polaco (Polish Zloty)
    SGD - Dólar de Singapur (Singapore Dollar)
    THB - Baht tailandés (Thai Baht)
    TRY - Lira turca (Turkish Lira)
    USD - Dólar estadounidense (United States Dollar)
    ZAR - Rand sudafricano (South African Rand)

In [15]:
df['employee_residence'].unique() 
# No hay errores de formato ni tipeo para esta columna

array(['CL', 'HU', 'JP', 'IN', 'IL', 'NO', 'TH', 'PH', 'CA', 'MX', 'US',
       'ZA', 'HK', 'TR', 'GB', 'QA', 'AU', 'FR', 'DK', 'CH', 'DE', 'ES',
       'EG', 'UA', 'RU', 'PL', 'NZ', 'BE', 'NG', 'PR', 'MY', 'BR', 'SA',
       'IE', 'SG', 'TN', 'PT', 'NL', 'CO', 'SE', 'IT', 'CN', 'BA', 'AE',
       'MD', 'LT', 'DO', 'UZ', 'MU', 'FI', 'IR', 'DZ', 'IQ', 'JE', 'AR',
       'HR', 'SI', 'LB', 'AT', 'VN', 'KE', 'GR', 'BG', 'MT', 'LV', 'BO',
       'CZ', 'RO', 'CY', 'GE', 'GH', 'KR', 'EE', 'PK', 'KW', 'AM', 'LU',
       'AD', 'OM', 'AS', 'CF', 'UG', 'PE', 'RS', 'CR', 'HN', 'EC', 'ID'],
      dtype=object)

#### Descripción países
    AD - Andorra
    AE - Emiratos Árabes Unidos (United Arab Emirates)
    AF - Afganistán (Afghanistan)
    AG - Antigua y Barbuda (Antigua and Barbuda)
    AI - Anguila (Anguilla)
    AL - Albania
    AM - Armenia
    AO - Angola
    AR - Argentina
    AS - Samoa Americana (American Samoa)
    AT - Austria
    AU - Australia
    AW - Aruba
    AX - Islas Åland (Åland Islands)
    AZ - Azerbaiyán (Azerbaijan)
    BA - Bosnia y Herzegovina (Bosnia and Herzegovina)
    BB - Barbados
    BD - Bangladés (Bangladesh)
    BE - Bélgica (Belgium)
    BF - Burkina Faso
    BG - Bulgaria
    BH - Bahréin (Bahrain)
    BI - Burundi
    BJ - Benín (Benin)
    BL - San Bartolomé (Saint Barthélemy)
    BM - Bermudas (Bermuda)
    BN - Brunéi (Brunei)
    BO - Bolivia
    BQ - Bonaire, San Eustaquio y Saba (Bonaire, Sint Eustatius and Saba)
    BR - Brasil (Brazil)
    BS - Las Bahamas (Bahamas)
    BT - Bután (Bhutan)
    BV - Isla Bouvet (Bouvet Island)
    BW - Botsuana (Botswana)
    BY - Bielorrusia (Belarus)
    BZ - Belice (Belize)
    CA - Canadá (Canada)
    CC - Islas Cocos (Cocos (Keeling) Islands)
    CD - República Democrática del Congo (Democratic Republic of the Congo)
    CF - República Centroafricana (Central African Republic)
    CG - República del Congo (Republic of the Congo)
    CH - Suiza (Switzerland)
    CI - Costa de Marfil (Ivory Coast)
    CK - Islas Cook (Cook Islands)
    CL - Chile
    CM - Camerún (Cameroon)
    CN - China
    CO - Colombia
    CR - Costa Rica
    CU - Cuba
    CV - Cabo Verde (Cape Verde)
    CW - Curazao (Curaçao)
    CX - Isla Christmas (Christmas Island)
    CY - Chipre (Cyprus)
    CZ - Chequia (Czech Republic)
    DE - Alemania (Germany)
    DJ - Yibuti (Djibouti)
    DK - Dinamarca (Denmark)
    DM - Dominica
    DO - República Dominicana (Dominican Republic)
    DZ - Argelia (Algeria)
    EC - Ecuador
    EE - Estonia
    EG - Egipto (Egypt)
    EH - Sahara Occidental (Western Sahara)
    ER - Eritrea
    ES - España (Spain)
    ET - Etiopía (Ethiopia)
    FI - Finlandia (Finland)
    FJ - Fiyi (Fiji)
    FM - Estados Federados de Micronesia (Federated States of Micronesia)
    FO - Islas Feroe (Faroe Islands)
    FR - Francia (France)
    GA - Gabón (Gabon)
    GB - Reino Unido (United Kingdom)
    GD - Granada (Grenada)
    GE - Georgia
    GF - Guayana Francesa (French Guiana)
    GG - Guernsey
    GH - Ghana
    GI - Gibraltar
    GL - Groenlandia (Greenland)
    GM - Gambia
    GN - Guinea
    GP - Guadalupe (Guadeloupe)
    GQ - Guinea Ecuatorial (Equatorial Guinea)
    GR - Grecia (Greece)
    GT - Guatemala
    GU - Guam
    GW - Guinea-Bisáu (Guinea-Bissau)
    GY - Guyana
    HK - Hong Kong
    HM - Islas Heard y McDonald (Heard Island and McDonald Islands)
    HN - Honduras
    HR - Croacia (Croatia)
    HT - Haití (Haiti)
    HU - Hungría (Hungary)
    ID - Indonesia
    IE - Irlanda (Ireland)
    IL - Israel
    IM - Isla de Man (Isle of Man)
    IN - India
    IO - Territorio Británico del Océano Índico (British Indian Ocean Territory)
    IQ - Irak (Iraq)
    IR - Irán (Iran)
    IS - Islandia (Iceland)
    IT - Italia (Italy)
    JE - Jersey
    JM - Jamaica
    JO - Jordania (Jordan)
    JP - Japón (Japan)
    KE - Kenia (Kenya)
    KG - Kirguistán (Kyrgyzstan)
    KH - Camboya (Cambodia)
    KI - Kiribati
    KM - Comoras (Comoros)
    KN - San Cristóbal y Nieves (Saint Kitts and Nevis)
    KP - Corea del Norte (North Korea)
    KR - Corea del Sur (South Korea)
    KW - Kuwait
    KY - Islas Caimán (Cayman Islands)
    KZ - Kazajistán (Kazakhstan)
    LA - Laos
    LB - Líbano (Lebanon)
    LC - Santa Lucía (Saint Lucia)
    LI - Liechtenstein
    LK - Sri Lanka
    LR - Liberia
    LS - Lesoto (Lesotho)
    LT - Lituania (Lithuania)
    LU - Luxemburgo (Luxembourg)
    LV - Letonia (Latvia)
    LY - Libia (Libya)
    MA - Marruecos (Morocco)
    MC - Mónaco (Monaco)
    MD - Moldavia (Moldova)
    ME - Montenegro
    MF - San Martín (Saint Martin)
    MG - Madagascar
    MH - Islas Marshall (Marshall Islands)
    MK - Macedonia del Norte (North Macedonia)
    ML - Malí (Mali)
    MM - Birmania (Myanmar)
    MN - Mongolia
    MO - Macao (Macao)
    MP - Islas Marianas del Norte (Northern Mariana Islands)
    MQ - Martinica (Martinique)
    MR - Mauritania
    MS - Montserrat
    MT - Malta
    MU - Mauricio (Mauritius)
    MV - Maldivas (Maldives)
    MW - Malaui (Malawi)
    MX - México (Mexico)
    MY - Malasia (Malaysia)
    MZ - Mozambique
    NA - Namibia
    NC - Nueva Caledonia (New Caledonia)
    NE - Níger (Niger)
    NF - Isla Norfolk (Norfolk Island)
    NG - Nigeria
    NI - Nicaragua
    NL - Países Bajos (Netherlands)
    NO - Noruega (Norway)
    NP - Nepal
    NR - Nauru
    NU - Niue
    NZ - Nueva Zelanda (New Zealand)
    OM - Omán (Oman)
    PA - Panamá (Panama)
    PE - Perú (Peru)
    PF - Polinesia Francesa (French Polynesia)
    PG - Papúa Nueva Guinea (Papua New Guinea)
    PH - Filipinas (Philippines)
    PK - Pakistán (Pakistan)
    PL - Polonia (Poland)
    PM - San Pedro y Miquelón (Saint Pierre and Miquelon)
    PN - Islas Pitcairn (Pitcairn Islands)
    PR - Puerto Rico (Puerto Rico)
    PT - Portugal
    PW - Palaos (Palau)
    PY - Paraguay
    QA - Catar (Qatar)
    RE - Reunión (Réunion)
    RO - Rumania (Romania)
    RS - Serbia
    RU - Rusia (Russia)
    RW - Ruanda (Rwanda)
    SA - Arabia Saudita (Saudi Arabia)
    SB - Islas Salomón (Solomon Islands)
    SC - Seychelles
    SD - Sudán (Sudan)
    SE - Suecia (Sweden)
    SG - Singapur (Singapore)
    SH - Santa Elena (Saint Helena)
    SI - Eslovenia (Slovenia)
    SJ - Svalbard y Jan Mayen (Svalbard and Jan Mayen)
    SK - Eslovaquia (Slovakia)
    SL - Sierra Leona (Sierra Leone)
    SM - San Marino (San Marino)
    SN - Senegal
    SO - Somalia
    SR - Surinam (Suriname)
    SS - Sudán del Sur (South Sudan)
    ST - Santo Tomé y Príncipe (São Tomé and Príncipe)
    SV - El Salvador
    SX - Sint Maarten
    SY - Siria (Syria)
    SZ - Suazilandia (Eswatini)
    TC - Islas Turcas y Caicos (Turks and Caicos Islands)
    TD - Chad
    TF - Territorios Australes Franceses (French Southern Territories)
    TG - Togo
    TH - Tailandia (Thailand)
    TJ - Tayikistán (Tajikistan)
    TK - Tokelau
    TL - Timor Oriental (Timor-Leste)
    TM - Turkmenistán (Turkmenistan)
    TN - Túnez (Tunisia)
    TO - Tonga
    TR - Turquía (Turkey)
    TT - Trinidad y Tobago (Trinidad and Tobago)
    TV - Tuvalu
    TZ - Tanzania
    UA - Ucrania (Ukraine)
    UG - Uganda
    UM - Islas Ultramarinas Menores de Estados Unidos (U.S. Minor Outlying Islands)
    US - Estados Unidos (United States)
    UY - Uruguay
    UZ - Uzbekistán (Uzbekistan)
    VA - Ciudad del Vaticano (Vatican City)
    VC - San Vicente y las Granadinas (Saint Vincent and the Grenadines)
    VE - Venezuela
    VG - Islas Vírgenes Británicas (British Virgin Islands)
    VI - Islas Vírgenes de los EE.UU. (U.S. Virgin Islands)
    VN - Vietnam
    VU - Vanuatu
    WF - Wallis y Futuna (Wallis and Futuna)
    WS - Samoa (Samoa)
    YE - Yemen
    YT - Mayotte
    ZA - Sudáfrica (South Africa)
    ZM - Zambia
    ZW - Zimbabue (Zimbabwe)

In [16]:
df['company_location'].unique()
# No hay errores de formato ni tipeo para esta columna

array(['CL', 'US', 'HU', 'JP', 'IN', 'ID', 'DE', 'CH', 'IL', 'SG', 'AS',
       'NO', 'TH', 'PH', 'CA', 'MX', 'ZA', 'HK', 'TR', 'GB', 'QA', 'AU',
       'DK', 'ES', 'FR', 'EG', 'UA', 'RU', 'PL', 'NZ', 'NG', 'PR', 'BR',
       'SA', 'IE', 'NL', 'CO', 'SE', 'BA', 'AE', 'LT', 'PT', 'MU', 'CZ',
       'FI', 'IR', 'DZ', 'IQ', 'CN', 'AR', 'HR', 'SI', 'LB', 'AT', 'VN',
       'KE', 'GR', 'BE', 'MT', 'LV', 'IT', 'GI', 'RO', 'EE', 'LU', 'KR',
       'AM', 'AD', 'OM', 'BS', 'CF', 'PK', 'MY', 'GH', 'HN', 'MD', 'EC'],
      dtype=object)

In [17]:
df['company_size'].unique()

array(['L', 'S', 'M'], dtype=object)

#### Descripcion tamaño empresas
S = Small (Pequeña)
M = Medium (Media)
L = Larga (Grande)